src/context/WalletProvider/NewWalletViews/wallets/mipd/MipdBody.tsx

```diff
@@ -0,0 +1,219 @@
+      } else {
+        setError(
+          translate('walletProvider.mipd.errors.unknown', {
+            name: maybeMipdProvider?.info.name ?? 'MetaMask',
+          }),
+        )
+        history.push('/metamask/failure')
+      }
+    }
+    setIsLoading(false)
+  }, [
+    dispatch,
+    getAdapter,
+    history,
+    isMetaMaskMobileWebView,
+    localWallet,
+    maybeMipdProvider?.info.name,
+    maybeMipdProvider?.info.rdns,
+    setError,
+    setIsLoading,
+    translate,
+  ])
+
+  const handleMetamaskRedirect = useCallback(() => {
+    const METAMASK_DEEP_LINK_BASE_URL = 'https://metamask.app.link'
+    // This constructs the MetaMask deep-linking target from the currently-loaded
+    // window.location. The port will be blank if not specified, in which case it
+    // should be omitted.
+    const mmDeeplinkTarget = [window.location.hostname, window.location.port]
+      .filter(x => !!x)
+      .join(':')
+    return window.location.assign(`${METAMASK_DEEP_LINK_BASE_URL}/${mmDeeplinkTarget}`)
+  }, [])
+
+  const connectBodyTranslation: [string, InterpolationOptions] = useMemo(
+    () => ['walletProvider.mipd.connect.body', { name: maybeMipdProvider?.info.name }],
+    [maybeMipdProvider],
+  )
+
+  if (!maybeMipdProvider) return null
+
+  if (isMobile && !isMetaMaskMobileWebView && rdns === METAMASK_RDNS) {
+    return (
+      <Flex direction='column' alignItems='center' justifyContent='center' height='full' gap={6}>
+        <Image src={maybeMipdProvider.info.icon} boxSize='64px' />
+        <Text fontSize='xl' translation='walletProvider.metaMask.redirect.header' mb={4} />
+        <Text
+          color='gray.500'
+          translation='walletProvider.metaMask.redirect.body'
+          textAlign='center'
+          mb={6}
+        />
+        <Button
+          width='200px'
+          colorScheme='blue'
+          onClick={handleMetamaskRedirect}
+          isDisabled={isLoading}
+        >
+          <Text translation='walletProvider.metaMask.redirect.button' />
+        </Button>
+        {error && (
+          <Alert status='info' mt={4}>
+            <AlertIcon />
+            <AlertDescription>
+              <Text translation={error} />
+            </AlertDescription>
+          </Alert>
+        )}
+      </Flex>
+    )
+  }
+
+  return (
+    <Flex direction='column' alignItems='center' justifyContent='center' height='full' gap={6}>
+      <Image src={maybeMipdProvider.info.icon} boxSize='64px' />
+      <CText fontSize='xl'>Pair {maybeMipdProvider.info.name}</CText>
+      <Text color='gray.500' translation={connectBodyTranslation} textAlign='center' />
+
+      {error && (
+        <Alert status='info'>
+          <AlertIcon />
+          <AlertDescription>
+            <Text translation={error} />
+          </AlertDescription>
+        </Alert>
+      )}
+
+      <Button
+        width='200px'
+        colorScheme='blue'
+        isLoading={isLoading}
+        loadingText='Pairing'
```

Comment:
Is it translated?

Reply:
c911cb8cee

<img width="1055" alt="image" src="https://github.com/user-attachments/assets/d2284b44-1dbb-4272-a853-0ed94d8cad86" />